Bottom: 11a95c9592045318cd25a0fa4631ccbb0ea056fc
Top:    8bc69cc85f5ae9c4288d7ebda349e8b41ca26782
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2013-01-03 04:09:41 +0000

find.c: Kill "SetThing" with fire


---

diff --git a/src/find.c b/src/find.c
index 0f85f3e..18385d7 100644
--- a/src/find.c
+++ b/src/find.c
@@ -401,15 +401,11 @@ PinLineIntersect (PinType *PV, LineType *Line)
 bool
 SetThing (int type, void *ptr1, void *ptr2, void *ptr3)
 {
+  assert (! (type == PIN_TYPE && ptr1 == NULL));
   thing_ptr1 = ptr1;
   thing_ptr2 = ptr2;
   thing_ptr3 = ptr3;
   thing_type = type;
-  if (type == PIN_TYPE && ptr1 == NULL)
-    {
-      thing_ptr1 = ptr3;
-      thing_type = VIA_TYPE;
-    }
   return true;
 }
